cypress/integration/task-2.js

```diff
@@ -1,94 +1,39 @@
 describe('Task 2', function() {
 
-    it('T-2: Users should be able to log in with "User" role and see appropriate tabs', function () {
-
-        const user = "TestCon User 6"
-        const role = "User"
-
-        cy.login(user, role)
-
-        //"User" role asserts
-        cy.url().should('include', '/time-logging')
-        cy.get('.page__title').contains('Timesheets')
-        cy.get('.calendar').should('be.visible')
-        cy.get('.tile.form').should('be.visible')
-        cy.get('.user-info__title').contains('TestCon User 6')
-        cy.get('.main-nav').find('li').should('have.length', 1)
-        cy.get('.main-nav').contains("Time Logging").should('have.css', 'color','rgb(64, 76, 237)')
+    it('T-2: All user roles can log in and see appropriate tabs', function () {
         
-    })
-
-    it('T-2: Users should be able to log in with "Team Lead" role and see appropriate tabs', function () {
-
+        let roles = ["User", "Team Lead", "Manager", "Accountant", "Admin"]
+        const nav_length = {
+            "User" : 1,
+            "Team Lead" : 2,
+            "Manager" : 5,
+            "Accountant" : 5,
+            "Admin" : 6,
+        }
         const user = "TestCon User 6"
-        const role = "Team Lead"
+        let i = 0
 
-        cy.login(user, role)
+        while (roles[i]){
```

Comment:
The idea is correct, but I think it would be more beneficial to extract the loop outside of the `it` test block, so that it would iterate through all the roles and launch each test separately instead of launching a single test and looping inside it :)

Reply:
I have tried that one already but couldn't make it work. Wasn't able to access the values of the array inside the "it" test case.

Reply:
foreach is a very good idea, thanks!

Reply:
Hm, it should be possible. Put the object array into the `describe` block, and then iterate through it with `it` block statements inside the loop. If you still run into problems, let me know and we'll figure it out :)